### PR TITLE
run-browser: Replaced PhantomJS with Puppeteer

### LINKED ISSUE #180: run-browser is an unmaintained project
Hello Team,

The dependency run-browser used by the xhr for testing the package is now unmaintained.
 
Further, the browser dependency used by run-browser is also out of date i.e the phantomjs browser.

I am curious to know that if anyone here has considered using a maintained alternative to run-browser for xhr.

Please do share your thoughts on this.

### PR DESCRIPTION
**Pull request by ossdev07**
PhantomJS is an unmaintained project, replaced phantomjs dependency with the puppeteer.

Closes https://github.com/naugtur/xhr/issues/180

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -69,10 +67,6 @@ function createHandler(filename, reports, phantom, mockserverHandler) {
         files = files.map(normalizePath);
         files.unshift(path.join(__dirname, '/lib/override-log.js'));
 
-        if (phantom) {
-          files.unshift(path.join(__dirname, '/lib/phantom-function-bind-shim.js'));
-        }
-
```
Comment (naugtur) [inline index.js:75]: oh wow, that's great :)

File: index.js [hunk 2]
```diff
@@ -69,10 +67,6 @@ function createHandler(filename, reports, phantom, mockserverHandler) {
         files = files.map(normalizePath);
         files.unshift(path.join(__dirname, '/lib/override-log.js'));
```
Comment (naugtur) [inline index.js:68]: I think there might be a better way to get logs from puppeteer than this, but it's ok to leave it as is for now.

**Review by naugtur — APPROVED** [review]
looks good to me.

**Comment by ossdev07** [comment]
Thanks, @naugtur for reviewing the patch, if it looks good to you when can I expect this PR to be merged?

**Comment by ossdev07** [comment]
@naugtur if the patch looks good to you can you please merge this pull request.

**Comment by naugtur** [comment]
Sorry, busy week with family.
I can merge, but I think there's more to be done here. I shouldn't ld be more available next week.

**Comment by ossdev07** [comment]
Thanks for merging this @naugtur, I have raised another pull request to resolve the failing test cases with puppeteer in xhr, Please have a look: https://github.com/naugtur/xhr/pull/184

**Comment by ossdev07** [comment]
@naugtur it will be really helpful if you could make a release with the changes in this PR.

**Pull request merged**